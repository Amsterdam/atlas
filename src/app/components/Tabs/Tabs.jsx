import React from 'react';
import PropTypes from 'prop-types';

const Tabs = ({ children: childrenProp, noResults, tip, currentTab }) => {
  let childIndex = 0;
  const children = React.Children.map(childrenProp, (child) => {
    if (!React.isValidElement(child)) {
      return null;
    }

    const childValue = child.props.label === undefined ? childIndex : child.props.label;
    const selected = childValue === currentTab;

    childIndex += 1;
    return React.cloneElement(child, {
      isCurrentTab: selected
    });
  });
  return (
    <div>
      <div className="o-tabs qa-tabs u-margin__bottom--2">
        <ul>
          {children}
        </ul>
      </div>

      {noResults ? (
        <div>
          <div className="u-margin__bottom--1">Geen resultaten van deze soort</div>
          <div>Tip: {tip}</div>
        </div>
      ) : null}
    </div>
  );
};

Tabs.defaulProps = {
  currentTab: false
};

Tabs.propTypes = {
  children: PropTypes.node,
  tip: PropTypes.string,
  currentTab: PropTypes.oneOfType([PropTypes.string, PropTypes.number, PropTypes.bool]),
  noResults: PropTypes.bool
};

export default Tabs;
