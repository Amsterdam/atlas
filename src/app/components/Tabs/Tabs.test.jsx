import React from 'react';
import { shallow } from 'enzyme';
import Tabs from './Tabs';

describe('Tabs', () => {
  it('should render the component', () => {
    const props = {
      currentTab: 'Dataset'
    };

    const component = shallow(<Tabs {...props} >
      <li key="Dataset">
        <span>This is a tab child</span>
      </li>
      <li key="Data" label="test-label">
        <span>This is a tab child</span>
      </li>
    </Tabs>);
    expect(component).toMatchSnapshot();
  });

  it('should handle component when there are invalid children', () => {
    const props = {
      currentTab: 'Dataset'
    };

    const component = shallow(<Tabs {...props} >
      [
        {React.createElement(jest.fn)},  // invalid child
        <li key="Data" label="test-label">
          <span>This is a tab child</span>
        </li>
      ]
    </Tabs>);
    expect(component).toMatchSnapshot();
  });
});
