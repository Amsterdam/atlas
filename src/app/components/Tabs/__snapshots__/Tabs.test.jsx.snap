// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tabs should handle component when there are invalid children 1`] = `
<div>
  <div
    className="o-tabs qa-tabs u-margin__bottom--2"
  >
    <ul>
      <bound fn
        isCurrentTab={false}
        key=".1"
      />
      <li
        isCurrentTab={false}
        key=".$Data"
        label="test-label"
      >
        <span>
          This is a tab child
        </span>
      </li>
    </ul>
  </div>
</div>
`;

exports[`Tabs should render the component 1`] = `
<div>
  <div
    className="o-tabs qa-tabs u-margin__bottom--2"
  >
    <ul>
      <li
        isCurrentTab={false}
        key=".$Dataset"
      >
        <span>
          This is a tab child
        </span>
      </li>
      <li
        isCurrentTab={false}
        key=".$Data"
        label="test-label"
      >
        <span>
          This is a tab child
        </span>
      </li>
    </ul>
  </div>
</div>
`;
